content/use-cases/ebr.md

```diff
@@ -0,0 +1,175 @@
+>}}
+
+With the sources of data and their corresponding models, the next step
+is to write a [{{< abbr "BPMN" >}}]({{< relref "/how-to/bpmn" >}}) workflow to automatically transform the data and update the database.
+
+{{< notice "note" >}}
+You may want to break these steps into multiple parts.
+Or, for increased modularity, you can call another BPMN workflow with a [Call activity]({{< relref "how-to/bpmn/bpmn-elements">}}#call-activities).
+{{< /notice >}}
+
+The procedure is as follows:
+
+1. Subscribe to the topic for the appropriate data (for example `/lims/lab1`). If the data comes from certain equipment, you first need to [Connect a data source]({{< relref "how-to/publish-subscribe/connect-datasource" >}}).
+
+1. Transform with JSONata
+
+   Rhize has a built-in JSONata interpreter, which can filter and transform JSON.
+   Use a [JSONata service task]({{< relref "how-to/bpmn/bpmn-elements">}}#jsonata-transform) to map the data sources into the corresponding ISA-95 fields that you defined on the previous step.
+  
+   Use the output as a variable for the next step.
+   
+
+1. POST data with a graph mutation.
+
+   Use the Variable returned by the JSONata step to send a mutation to update the Graph database with the new fields.
+   To learn more, read the [Guide to GraphQL with Rhize]({{< relref "how-to/gql" >}}).
+
+In real BPMN workflows, you can dynamically create and assign fields as they enter the system.
+For example, this workflow creates a new material definition and material-definition version based on whether this object already exists.
+
+<a href="/images/bpmn/screenshot-rhize-bpmn-add-material-definition.png" target="_blank">
+{{< figure
+alt="Screenshot of a BPMN workflow that adds material only if it exists"
+src="/images/bpmn/screenshot-rhize-bpmn-add-material-definition.png"
+>}}
+</a>
+
+This step can involve multiple BPMN processes subscribing to different topics.
+As long as the incoming event data has a common association, for example through the `id` of the batch data and associated `JobResponse`, you can return all eBR fields in one GraphQL query&mdash;no recursive SQL joins needed.
+
+{{< figure
+alt="Multiple BPMN processes can be united in one batch"
+src="/images/ebr/diagram-rhize-inputs-for-ebr.png"
+width="75%"
+>}}
+
+### Query the DB with the eBR fields
+
+After the batch finishes, use a [GraphQL query]({{< relref "how-to/gql/query-mutate-subscribe" >}}) to receive all relevant batch data.
+You only need to write one request, and it returns exactly the data you specify.
+
+{{< figure
+alt="Diagram showing how a query makes an ebr"
+src="/images/ebr/diagram-rhize-make-ebr-query.png"
+width="55%"
+>}}
+
+Our experience has shown us that the query is about 300 lines. Here's a
+small, generic snippet of how it looks:
+
+{{< expandable title="Snippet of a makeEbr query" >}}
+```graphql
+query makeEbr ($filter: JobOrderFilter) {
```

Comment:
This snippet is of the planned execution, not the actual. I'd suggest changing it to:

```graphql
query makeEbr ($filter: JobResponseFilter) {
  queryJobResponse(filter: $filter) {
    EXAMPLE_id: id
    description
    matActualProduced: materialActual(filter:{materialUse: { eq:Produced }}){
      id
      material: materialDefinitionVersion{
        id
      }
      quantity
      quantityUoM {
        id
      }
    }
  ## More EBR fields
  }
}
```